### PR TITLE
Privacy Settings: Hide for Unlaunched Sites

### LINKED ISSUE #46629: Hide privacy settings for unlaunched sites
pbxlJb-rr-p2#comment-626

> I’m not sure if users would find this as an issue, but the grayed out box in the settings page has the public option selected by default. I get that’s exactly what the V2 attempts to do, but from a user’s perspective who doesn’t know all of that, it looks like by default the site is public.
The text next to the Launch site button says that the site is “private”, and yet the “Public” radio option is selected.
 
screenshot-2020-10-20-at-7 17 49-pm

### PR DESCRIPTION
**Pull request by Aurorum**
#### Changes proposed in this Pull Request

* Hides the Privacy Settings for unlaunched sites instead of just disabling it

#### Testing instructions

Visit `/settings/general/` and scroll down to the Privacy Settings. There's a couple of cases to verify. Firstly, nothing should change for launched sites. 

For unlaunched sites who haven't verified their email, that prompt should still appear with the "Launch" button.

**Before:**
Screenshot 2020-10-25 at 14 26 22

**After:**
Screenshot 2020-10-25 at 14 26 33

For unlaunched sites who have verified the email, the "Launch" button should still be available. 

**Before:**
Screenshot 2020-10-25 at 14 26 08

 **After:**
Screenshot 2020-10-25 at 14 26 13

cc @roo2, @scruffian

Fixes #46629

### CODE REVIEW

File: client/my-sites/site-settings/form-general.jsx [hunk 1]
```diff
@@ -653,35 +653,16 @@ export class SiteSettingsFormGeneral extends Component {
...
-		e.target.blur();
-	};
-
 	privacySettingsWrapper() {
 		if ( this.props.isUnlaunchedSite ) {
-			if ( this.props.needsVerification ) {
-				return (
-					<EmailVerificationGate>
-						{ this.renderLaunchSite() }
-						{ this.privacySettings() }
```
Comment (roo2) [inline client/my-sites/site-settings/form-general.jsx]: we've been working on this as part of coming soon, so my first instinct is to say we should feature toggle this behind the coming-soon-v2 feature toggle e.g. by doing something like:
```
	 { ! config.isEnabled( 'coming-soon-v2' ) && this.privacySettings() }
```

But hmm maybe it's fine to just make this change for everyone, that would certainly make the implementation simpler and allow us to get that refactoring in. I've reached out to our designer on if we want this change to affect all sites immediately or just the ones that are using coming soon v2

**Review by roo2 — COMMENTED** [review]
Thank you for jumping in and doing this!

This issue came up originally because we've been using a new simplified coming soon mode. but we had a bug with this new coming soon mode, where sites where shown as public on the settings page before they were launched. So we suggested to just hide these settings before a site is launched to avoid the issue. 

But I just need to confirm that this change makes sense for all sites or if it's just a fix to make our life easier for these "Coming soon v2" sites

Again, thanks for getting in and creating this PR! 😄

The code looks good and works as intended

**Comment by roo2** [comment]
hey I'm just trying to get our unit tests running, would you please rebase this onto master and I'll kick them off again? 🙏

**Comment by roo2** [comment]
bump, to get this in we need to rebase it onto master with e.g.
```
git fetch
git rebase origin/master # if you've called the automattic calypso repo "origin"
git push --force-with-lease
```

Then I'll be able to run our unit tests against this and get it merged!

Thank you 🙏

**Comment by Aurorum** [comment]
Sorry for being so quiet lately, and for leaving this so long

I've always had issues using the command line like that, but I've tried to rebase. It seems some of the unit tests always fail for me since this is an external PR!

**Comment by roo2** [comment]
hey, sorry I missed the ping back in November! Would you still like to get this in? You'll have to rebase onto the `trunk` branch which we now use for our calypso releases like so:

```
git fetch
git rebase origin/trunk # if you've called the automattic calypso repo "origin". We now use the trunk branch.
git push --force-with-lease
```

**Comment by Aurorum** [comment]
Hi - happy to still work on this! Are the new changes which I've made okay?

**Comment by Aurorum** [comment]
The tests appear to be failing.

https://github.com/Automattic/wp-calypso/blob/ed3e91545e431da92beeffcbe80289eadde067c0/client/my-sites/site-settings/test/form-general.jsx#L233-L250

@roo2 I'm not entirely certain, so just to check, should the Privacy Settings only be hidden for unlaunched sites that are not private, or should they be hidden for all unlaunched sites? (In which case, should the tests be changed?)

**Comment by roo2** [comment]
hey, thanks for working on this again! and yep it looks like that one test is failing.

[wp-calypso/client/my-sites/site-settings/test/form-general.jsx](https://github.com/Automattic/wp-calypso/blob/ed3e91545e431da92beeffcbe80289eadde067c0/client/my-sites/site-settings/test/form-general.jsx#L233-L250)

It looks like we can just remove that one test though because it tests the feature that we're removing.

Currently a different team has been experimenting with deactivating this settings page and using the wp-admin settings page on wordpress.com but I think in the long term we do still do want to use this settings page.

>should the Privacy Settings only be hidden for unlaunched sites that are not private, or should they be hidden for all unlaunched sites

They should be hidden for all unlaunched sites

>In which case, should the tests be changed?

I think just removing that one test is ok for now, but if you do want to add a test that asserts the options are not shown when the site is "coming soon", then that would be good 👍

**Comment by roo2** [comment]
@Aurorum would you mind if I take this work or implement it in my own PR, I think it is easier for me to get it through the automated tests etc. ( Thanks for working on this and I appreciate your contribution to Calypso! )

**Comment by Aurorum** [comment]
Yes, that's fine - thank you!

**Pull request closed without merging**